sale_by_packaging/tests/test_sale_only_by_packaging.py

```diff
@@ -0,0 +1,141 @@
+
+        with self.assertRaises(ValidationError):
+            order_line.write({"product_packaging_id": False})
+
+    def test_create_auto_fill_packaging(self):
+        """Check when the packaging should be set automatically on the line"""
+        # sell_only_by_packaging is default False
+        with Form(self.order) as so:
+            with so.order_line.new() as so_line:
+                so_line.product_id = self.product
+                so_line.product_uom_qty = self.packaging_tu.qty * 2
+        so_line = self.order.order_line[-1]
+        self.assertFalse(so_line.product_packaging_id)
+        self.assertFalse(so_line.product_packaging_qty)
+
+        # If sell_only_by_packaging is set, a packaging should be automatically
+        # picked if possible
+        self.product.sell_only_by_packaging = True
+        with Form(self.order) as so:
+            with so.order_line.new() as so_line:
+                so_line.product_id = self.product
+                so_line.product_uom_qty = self.packaging_tu.qty * 2
+        so_line = self.order.order_line[-1]
+        self.assertTrue(so_line.product_packaging_id)
+        self.assertTrue(so_line.product_packaging_qty)
+        self.assertEqual(so_line.product_packaging_id, self.packaging_tu)
+        self.assertEqual(so_line.product_packaging_qty, 2)
+
+        # If qty does not match a packaging qty, an exception should be raised
+        with self.assertRaises(ValidationError):
+            with Form(self.order) as so:
+                with so.order_line.new() as so_line:
+                    so_line.product_id = self.product
+                    so_line.product_uom_qty = 2
+
+    @mute_logger("odoo.tests.common.onchange")
+    def test_convert_packaging_qty(self):
+        """
+        Test if the function _convert_packaging_qty is correctly applied
+        during SO line create/edit and if qties are corrects.
+        :return:
+        """
+        self.product.sell_only_by_packaging = True
+        packaging = self.packaging_tu
+        # For this step, the qty is not forced on the packaging so nothing
+        # should happens if the qty doesn't match with packaging multiple.
+        with Form(self.order) as sale_order:
+            with sale_order.order_line.edit(0) as so_line:
+                so_line.product_packaging_id = packaging
+                so_line.product_uom_qty = 12
+                self.assertAlmostEqual(
+                    so_line.product_uom_qty, 12, places=self.precision
+                )
+                so_line.product_uom_qty = 10
+                self.assertAlmostEqual(
+                    so_line.product_uom_qty, 10, places=self.precision
+                )
+                so_line.product_uom_qty = 36
+                self.assertAlmostEqual(
+                    so_line.product_uom_qty, 36, places=self.precision
+                )
+                so_line.product_uom_qty = 10
+                so_line.product_packaging_id = packaging
```

Comment:
This line breaks the test.

```python
2023-06-19 13:47:42,590 937136 ERROR 16-sbp odoo.addons.sale_by_packaging.tests.test_sale_only_by_packaging: FAIL: TestSaleProductByPackagingOnly.test_convert_packaging_qty
Traceback (most recent call last):
  File "/var/home/coopiteasy/cie/odoo16/src/odoo/odoo/tools/misc.py", line 782, in deco
    return func(*args, **kwargs)
  File "/var/home/coopiteasy/Projets/OCA/sale-workflow/sale_by_packaging/tests/test_sale_only_by_packaging.py", line 94, in test_convert_packaging_qty
    so_line.product_packaging_id = packaging
  File "/var/home/coopiteasy/cie/odoo16/src/odoo/odoo/tests/common.py", line 2216, in __setattr__
    assert descr is not None, "%s was not found in the view" % field
AssertionError: product_packaging_id was not found in the view
```

I'm not really sure why this happens, but as stated above, this line is broken _anyway_, because `product_packaging_id` is a computed field.